service/reducers/entities/files.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {FilesTypes, UsersTypes} from 'service/constants';
+
+function files(state = {}, action) {
+    switch (action.type) {
+    case FilesTypes.RECEIVED_FILES_FOR_POST: {
+        const filesById = action.data.reduce((filesMap, file) => {
+            return {...filesMap,
+                [file.id]: file
+            };
+        }, {});
+        return {...state,
+            ...filesById
+        };
+    }
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function fileIdsByPostId(state = {}, action) {
```

Comment:
@hmhealey Ah, now I recall the primary reasoning. Without the `fileIdsByPostId` reducer, we don't have a way to know in `loadFilesForPostIfNecessary` if we've requested the post's files yet. I think we would end up fetching the files every time the `FileAttachmentList` remounted.